ui/src/plugins/dev.perfetto.RecordTraceV2/pages/target_selection_page.ts

```diff
@@ -53,40 +71,131 @@ export function targetSelectionPage(recMgr: RecordingManager): RecordSubpage {
     },
     async deserialize(state: RecordPluginSchema) {
       recMgr.autoOpenTraceWhenTracingEnds = state.autoOpenTrace;
-      if (state.target.platformId === undefined) return;
-      recMgr.setPlatform(state.target.platformId);
+
+      // Restore platform selection
+      if (state.target.platformId !== undefined) {
+        recMgr.setPlatform(state.target.platformId);
+      }
+
+      // Restore provider selection
       const prov = recMgr.getProvider(state.target.transportId ?? '');
-      if (prov === undefined) return;
-      await recMgr.setProvider(prov);
-      if (state.target.targetId === undefined) return;
-      for (const target of await recMgr.listTargets()) {
-        if (target.id === state.target.targetId) {
-          await recMgr.setTarget(target);
+      if (prov !== undefined) {
+        await recMgr.setProvider(prov);
+      }
+
+      // Restore target selection
```

Comment:
As discussed offline, rather than doing all this reverse matching, i'd remember the preset name, and re apply it.
If the preset name is the "custom" one then you reload the saved probes rather than reloading the preset as if the user did "blank" and then clicked on the quick start config